src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -257,6 +258,97 @@ internal string GetString(int index, JsonTokenType expectedType)
             return lastString;
         }
 
+        internal bool TextEquals(int index, ReadOnlySpan<char> otherText, bool isPropertyName)
+        {
+            CheckNotDisposed();
+
+            int matchIndex = isPropertyName ? index - DbRow.Size : index;
+
+            (int lastIdx, string lastString) = _lastIndexAndString;
+        
+            if (lastIdx == matchIndex)
+            {
+                return otherText.SequenceEqual(lastString);
+            }
+
+            Span<byte> otherUtf8Text;
+            byte[] otherUtf8TextArray = null;
+
+            int length = otherText.Length;
```

Comment:
Same concern here/elsewhere we have transocding. I believe we still need to do the 3x multiply.